packages/@react-spectrum/s2/src/ActionBar.tsx

```diff
@@ -0,0 +1,233 @@
+    isEmphasized: 'transparent',
+    forcedColors: 'ButtonBorder'
+  },
+  paddingX: 8,
+  paddingY: 12,
+  display: 'flex',
+  gap: 16,
+  alignItems: 'center',
+  position: {
+    isInContainer: 'absolute'
+  },
+  bottom: 8,
+  insetStart: 8,
+  '--insetEnd': {
+    type: 'insetEnd',
+    value: 8
+  },
+  width: {
+    default: 'full',
+    isInContainer: 'auto'
+  },
+  marginX: 'auto',
+  maxWidth: 960,
+  animation: {
+    isInContainer: slideIn,
+    isExiting: slideOut
+  },
+  animationDuration: 200
+});
+
+export interface ActionBarProps extends SlotProps {
+  /** A list of ActionButtons to display. */
+  children: ReactNode,
+  /** Whether the ActionBar should be displayed with a emphasized style. */
+  isEmphasized?: boolean,
+  /** The number of selected items that the ActionBar is currently linked to. If 0, the ActionBar is hidden. */
+  selectedItemCount?: number | 'all',
+  /** Handler that is called when the ActionBar clear button is pressed. */
+  onClearSelection?: () => void,
+  /** A ref to the scrollable element the ActionBar appears above. */
+  scrollRef?: RefObject<HTMLElement | null>
+}
+
+export const ActionBarContext = createContext<ContextValue<Partial<ActionBarProps>, DOMRefValue<HTMLDivElement>>>(null);
+
+export const ActionBar = forwardRef(function ActionBar(props: ActionBarProps, ref: DOMRef<HTMLDivElement>) {
+  [props, ref] = useSpectrumContextProps(props, ref, ActionBarContext);
+  let domRef = useDOMRef(ref);
+
+  let isOpen = props.selectedItemCount !== 0;
+  let isExiting = useExitAnimation(domRef, isOpen && props.scrollRef != null);
+  if (!isOpen && !isExiting) {
+    return null;
+  }
+
+  return <ActionBarInner {...props} ref={domRef} isExiting={isExiting} />;
+});
+
+const ActionBarInner = forwardRef(function ActionBarInner(props: ActionBarProps & {isExiting: boolean}, ref: ForwardedRef<HTMLDivElement | null>) {
+  let {isEmphasized, selectedItemCount = 0, children, onClearSelection, isExiting} = props;
+  let stringFormatter = useLocalizedStringFormatter(intlMessages, '@react-spectrum/s2');
+
+  // Store the last count greater than zero so that we can retain it while rendering the fade-out animation.
+  let [lastCount, setLastCount] = useState(selectedItemCount);
+  if ((selectedItemCount === 'all' || selectedItemCount > 0) && selectedItemCount !== lastCount) {
+    setLastCount(selectedItemCount);
+  }
+    
+  // Measure the width of the collection's scrollbar and offset the action bar by that amount.
+  let scrollRef = props.scrollRef;
+  let [scrollbarWidth, setScrollbarWidth] = useState(0);
+  let updateScrollbarWidth = useCallback(() => {
+    let el = scrollRef?.current;
+    if (el) {
+      let w = el.offsetWidth - el.clientWidth;
+      setScrollbarWidth(w);
+    }
+  }, [scrollRef]);
+
+  useResizeObserver({
+    ref: scrollRef,
+    onResize: updateScrollbarWidth
+  });
+
+  let {keyboardProps} = useKeyboard({
+    onKeyDown(e) {
+      if (e.key === 'Escape') {
```

Comment:
continue propagation for everything else?

Reply:
Maybe we should have a higher level API for `useKeyboard` to make this automatic... could be based on key codes or something

```jsx
useKeyboard({
  Enter(e) {
    // Do whatever
   }
});

Reply:
it'd be nice if it also didn't handle Shift+Enter unless explicitly declared